### PR TITLE
Rename handlebars helpers and its utils

### PR DESCRIPTION
**Pull request by gkhunger-sf**
Rename  handlebars helpers and its utils based on Jeff's quip doc - https://salesforce.quip.com/JNnDAivz5cyK

### CODE REVIEW

File: src/generate/utils.ts [hunk 1]
```diff
@@ -228,22 +249,23 @@ export const getValue = <T>(name: model.ValueField<T>): string => {
...
+ * @param node - An AMF node
+ * @param propertyFilter - predicate to be used to filter properties
  *
- * @param dtoTypeModel - AMF model of the dto
- * @param propertyFilter - function to filter properties based on certain criteria
  * @returns The filtered list of properties
  */
 export const getFilteredProperties = (
-  dtoTypeModel: model.domain.NodeShape | null | undefined,
+  node: model.domain.NodeShape | null | undefined,
```
Comment (wjhsf) [inline src/generate/utils.ts:260]: Will `null` or `undefined` ever actually be passed to the function?

File: src/generate/handlebarsAmfHelpers.ts [hunk 2]
```diff
@@ -77,12 +79,13 @@ export const getReturnPayloadType = (
...
+ * Get data type of a property.
+ *
+ * @param property - An AMF property
  *
- * @param property - instance of model.domain.PropertyShape
- * @returns data type if defined in the property otherwise returns a default type
+ * @returns data type, if defined in the property, the default type otherwise
  */
-export const getPropertyDataType = (
+export const getTypeFromProperty = (
```
Comment (monkey-jeff) [inline src/generate/handlebarsAmfHelpers.ts:88]: How does a property differ from a parameter?

File: src/generate/handlebarsAmfHelpers.ts [hunk 3]
```diff
@@ -174,19 +182,20 @@ export const isOptionalProperty = (
...
 ): boolean => {
   return (
-    ramlTypeDefinition !== undefined &&
-    ramlTypeDefinition.closed !== undefined &&
-    ramlTypeDefinition.closed.value !== undefined &&
-    !ramlTypeDefinition.closed.value()
+    node !== undefined &&
+    node.closed !== undefined &&
+    node.closed.value !== undefined &&
+    !node.closed.value()
```
Comment (wjhsf) [inline src/generate/handlebarsAmfHelpers.ts:198]:
```suggestion
    node?.closed?.value &&
    !node.closed.value()
```

File: src/generate/utils.ts (hunk 1 above)
Comment (gkhunger-sf) [reply src/generate/utils.ts:260]: Probably not. I didn't know the reason it was added for so didn't bother updating it.

File: src/generate/handlebarsAmfHelpers.ts (hunk 2 above)
Comment (gkhunger-sf) [reply src/generate/handlebarsAmfHelpers.ts:88]: The only difference is the kind of AMF element they operate on.
model.domain.PropertyShape vs model.domain.Parameter

File: src/generate/handlebarsAmfHelpers.ts (hunk 3 above)
Comment (gkhunger-sf) [reply src/generate/handlebarsAmfHelpers.ts:198]: Reverted this change since it made one of the tests fail.

Comment (wjhsf) [reply src/generate/handlebarsAmfHelpers.ts:198]:
```suggestion
    typeof node?.closed?.value === 'function' && !node.closed.value()
```
¯\\\_(ツ)_/¯

Comment (gkhunger-sf) [reply src/generate/handlebarsAmfHelpers.ts:198]: I trusted you once, I am not doing it again 😛

**Pull request merged**